packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -983,6 +999,48 @@ export class StatsAnalyzer extends EventsScope {
     }
   }
 
+  /**
+   * extracts the local Ip address from the statsResult object by looking at stats results candidates
+   * and matches that ID with the successful candidate pair. It looks at the type of local candidate it is
+   * and then extracts the IP address from the relatedAddress or address property based on conditions known in webrtc
+   * note, there are known incompatibilities and it is possible for this to set undefined, or for the IP address to be the public IP address
+   * for example, firefox does not set the relayProtocol, and if the user is behind a NAT it might be the public IP
+   * @private
+   * @param {string} successfulCandidatePairId - The ID of the successful candidate pair.
+   * @param {Object} candidates - the stats result candidates
+   * @returns {void}
+   */
+  extractAndSetLocalIpAddressInfoForDiagnostics = (
+    successfulCandidatePairId: string,
+    candidates: {[key: string]: Record<string, unknown>}
+  ) => {
+    let newIpAddress = '';
+    if (successfulCandidatePairId && !isEmpty(candidates)) {
+      const localCandidate = candidates[successfulCandidatePairId];
+      if (localCandidate) {
+        if (localCandidate.candidateType === 'host') {
+          // if it's a host candidate, use the address property - it will be the local IP
+          newIpAddress = `${localCandidate.address}`;
+        } else if (localCandidate.candidateType === 'prflx') {
+          // if it's a peer reflexive candidate and we're not using a relay (there is no relayProtocol set)
+          // then look at the relatedAddress - it will be the local
+          //
+          // Firefox doesn't populate the relayProtocol property
+          if (!localCandidate.relayProtocol) {
+            newIpAddress = `${localCandidate.relatedAddress}`;
+          } else {
+            // if it's a peer reflexive candidate and we are using a relay -
+            // in that case the relatedAddress will be the IP of the TURN server (Linus),
+            // so we can only look at the address, but it might be local IP or public IP,
+            // depending on if the user is behind a NAT or not
+            newIpAddress = `${localCandidate.address}`;
```

Comment:
I'm slightly confused by the existing code. I think this is the place where the statsResults.candidates are set (in parseCandidate):
![image](https://github.com/webex/webex-js-sdk/assets/7473332/91394c26-51ee-4362-aa5c-a22722e241da)
and this place sets `ipAddress` property not `address`, but when I run the sample app I can see both properties are set so seems like it is populated somewhere from the actual candidate. @mccarthytyler maybe you know where this is populated?

Reply:
dont you mean from here? https://github.com/webex/webex-js-sdk/pull/3357/files#diff-bfae05440410022d826a0f6d3ee83997e3471a619c6d7925db6da86a83597990R1085 or do you mean in existing existing code it's populated anyway?

There's no reference in the SDK to setting "it is populated somewhere from the actual candidate" so I'm not sure how that's happening, but when I ran it locally with the web client, those fields were not available so I added them using the code I referenced above

Reply:
ah, sorry, my mistake, I have 2 git clones of SDK and I had your PR checked out and running sample app from one of them, but when looking at the parseCandidate() function I was looking at the other clone